### PR TITLE
[good-logs] Package type definition added

### PR DESCRIPTION
**Pull request by ballyalley-o**
Please fill in this template.
 
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:

- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/good-logs/README.md [hunk 1]
```diff
@@ -0,0 +1,206 @@
+
```
Comment (jakebailey) [inline types/good-logs/README.md:1]: This readme will not be included in the package published to npm; I would remove this file.

**Review by jakebailey — CHANGES_REQUESTED** [review]
I don't think this should be added to DT; the package itself is written in TS and contains types:

image

You also own the package and can publish types with your package.

File: types/good-logs/README.md (hunk 1 above)
Comment (ballyalley-o) [reply types/good-logs/README.md:1]: Yes please, thank you.

**Comment by ballyalley-o** [comment]
> I don't think this should be added to DT; the package itself is written in TS and contains types:
> 
> image
> 
> You also own the package and can publish types with your package.

Yes it has the type defnition at first but then decided that its much better to add the type definition here instead. :-)

**Review by ballyalley-o — COMMENTED** [review]
Cheers!

**Review by ballyalley-o — COMMENTED** [review]
Deleted the Readme.md

**Comment by jakebailey** [comment]
> Yes it has the type defnition at first but then decided that its much better to add the type definition here instead. :-)

Then you should publish a new version without the types, because your package will override DT.

I'll also say that it's a little weird to have "but with typescript" on rhe repo, but it not have any TS at all?

But still, we typically do not want people who have permission to publish types to their packages to instead publish on DT. It's a worse experience for users.

**Comment by ballyalley-o** [comment]
> > Yes it has the type defnition at first but then decided that its much better to add the type definition here instead. :-)
> 
> Then you should publish a new version without the types, because your package will override DT.
> 
> I'll also say that it's a little weird to have "but with typescript" on rhe repo, but it not have any TS at all?
> 
> But still, we typically do not want people who have permission to publish types to their packages to instead publish on DT. It's a worse experience for users.

yes i did publish a new version without the types. 1.1.2 already has no types included, i dont know npm still showing the ts badge.

**Comment by jakebailey** [comment]
Your package still contains `"types": "index.d.ts"`.

**Comment by ballyalley-o** [comment]
Screenshot 2024-03-01 at 8 18 44 AM

**Comment by ballyalley-o** [comment]
> Thanks, though I still don't understand why you are trying to publish this on DT when you control the package. Users will have to install two packages, you'll need to send PRs to two different repos and get approvals (which will be tough as the sole author), etc. It's really not worth it.

I just want to keep them separate, so users can also use it for commonjs/module without downloading (smaller chunk of files.) the type definitions.  but also can be used for ts users by downloading the type-definition.

**Comment by ballyalley-o** [comment]
disregard the review request, i just accidentally clicked that. sorry.

**Comment by peterblazejewicz** [comment]
> I just want to keep them separate, so users can also use it for commonjs/module without downloading (smaller chunk of files.) the type definitions. but also can be used for ts users by downloading the type-definition.

Those gets zipped (compressed). Ends users gets compressed asset from NPM registry, which is decompressed. This is how package managers work mostly. (Have the same arguments at work some time ago). Overhead is negligible (we're talking plain text compression). thanks!

**Comment by ballyalley-o** [comment]
> > I just want to keep them separate, so users can also use it for commonjs/module without downloading (smaller chunk of files.) the type definitions. but also can be used for ts users by downloading the type-definition.
> 
> Those gets zipped (compressed). Ends users gets compressed asset from NPM registry, which is decompressed. This is how package managers work mostly. (Have the same arguments at work some time ago). Overhead is negligible (we're talking plain text compression). thanks!

i see, thanks peter.

File: types/good-logs/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+  /**
+   * Preset log type for connection status update in the console
+   * @param db - connection call
+   * @param isConnected - send the status of the db connection
+   */
+   function db(host: any, dbName: any, isConnected: boolean): void
+}
+
+
+export default goodlog
```
Comment (sandersn) [inline types/good-logs/index.d.ts:171]: good-log's index.js is commonjs. You should use the commonjs syntax `export = goodlog` instead.

**Review by sandersn — CHANGES_REQUESTED** [review]
You wrote good-logs yourself. You should ship these types in the good-log package as index.d.ts.

(The JS code looks like it was generated by Typescript, so you might get Typescript to generate the types for you too.)

File: types/good-logs/index.d.ts (hunk 2 above)
Comment (ballyalley-o) [reply types/good-logs/index.d.ts]: thank you for reviewing this, i made some changes, please review. thanks again @sandersn.

File: types/good-logs/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
+{
+    "compilerOptions": {
+        "module": "node16",
+         "lib": [
+            "es6",
+            "dom"
```
Comment (peterblazejewicz) [inline types/good-logs/tsconfig.json:6]:
```suggestion
            "es6"
```
you do not need this (DOM lib reference)

**Review by sandersn — CHANGES_REQUESTED** [review]
Please publish the types with good-log itself instead of here on DT.

**Pull request closed without merging**